lib/index.js

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) 2024 Digital Bazaar, Inc. All rights reserved.
+ */
+// RFC 9285 The Base45 Data Encoding
+// https://www.rfc-editor.org/rfc/rfc9285
+
+// base45 alphabet (RFC 9285)
+const encodeTable = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ $%*+-./:';
+const decodeTable = new Uint8Array(256);
+
+// build decode table, using 0xFF as invalid value.
+decodeTable.fill(0xFF);
+for(let i = 0; i < encodeTable.length; ++i) {
+  decodeTable[encodeTable.charCodeAt(i)] = i;
+}
+
+// decode one char and check validity
+function _decodeChar(c) {
+  const d = decodeTable[c];
+  if(d === 0xFF) {
+    throw new RangeError('Invalid input character.');
+  }
+  return d;
+}
+
+/**
+ * Encodes a Uint8Array to a string using the RFC 9285 base45 algorithm.
+ *
+ * @param {Uint8Array} input - The bytes to encode in a Uint8Array.
+ *
+ * @returns {string} The base45-encoded output string.
+ */
+export function encode(input) {
+  if(!(input instanceof Uint8Array)) {
+    throw new TypeError('"input" must be a Uint8Array.');
+  }
+  if(input.length === 0) {
+    return '';
+  }
+
+  let output = '';
+
+  const inputView = new DataView(input.buffer);
+  for(let i = 0; i < Math.floor(input.length / 2); ++i) {
+    const d = inputView.getUint16(i * 2);
+    output += encodeTable[d % 45];
+    output += encodeTable[Math.floor(d / 45) % 45];
+    output += encodeTable[Math.floor(d / (45 * 45)) % 45];
+  }
+  if(input.length % 2) {
+    const d = inputView.getUint8(input.length - 1);
+    output += encodeTable[d % 45];
+    output += encodeTable[Math.floor(d / 45) % 45];
+  }
```

Comment:
No idea what is more performant here and this can be dealt with later, but one idea is to create an array (potentially presized/preallocated) to store the chars from the encode table -- and then join it at the end with a single call... instead of building up the `output` string like this.

Reply:
Yeah, there are options.  Our other base libs use the basen file, which (roughly) grows an array as needed, then loops over that array to build string with `+=` while doing the alphabet mapping.  This version avoids the array and does mapping as it goes.  I imagine benchmarks would be particular to different js runtime optimizations.

Honestly I thought you'd ask about the use of DataView tricks rather than direct byte access.  I'm not sure which way is better.